src/System.Slices/src/System/Span.cs

```diff
@@ -299,27 +299,17 @@ public bool Equals(Span<T> other)
                 return false;
             }
 
-            Enumerator thisIt = this.GetEnumerator();
-            Enumerator otherIt = other.GetEnumerator();
-            while (true)
+            var nonBoxingComparer = EqualityComparer<T>.Default;
```

Comment:
It feels like it should exist on the span. I feel like this type without any useful methods will be quite useless. I'm not convinced about implementing all the interfaces although memory comparison, enumerators etc. should be here IMO